src/symbols-manual.js

```diff
@@ -238,6 +238,55 @@ Rocky.addManualSymbols = function(obj) {
     });
   };
 
+  obj.gpath_create = function(config) {
+    // handle case where config is an array
+    if (!('points' in config)) {
+      config = {points: config};
+    }
+    config.points = config.points.map(function(pt) {return obj.GPoint(pt);});
+    config.offset = obj.GPoint(config.offset || [0, 0]);
+    config.rotation = config.rotation || 0;
+    config.captureCPointer = function() {
+      // we're describing a GPath structure followed by the actual points
+      // typedef struct GPath {
+      //   uint32_t num_points; // 0
+      //   GPoint *points;      // 4
+      //   int32_t rotation;    // 8
+      //   GPoint offset;       // 12
+      // } GPath;               // 16+ actual points
+      var needed_bytes = 16 + this.points.length * 4;
+      var ptr = obj.module._malloc(needed_bytes);
+      obj.module.setValue(ptr + 0, this.points.length, 'i32');
```

Comment:
This is quite rough. I wonder whether or not we would be better off writing a few setters on the C side